### PR TITLE
Implement SFE Frontend

### PR DESCRIPTION
**Pull request by wottpal**
**Landing Page:**
stake-for-earth-6f34f86 vercel app_donate (2)

**Staking Subpage:**
stake-for-earth-6f34f86 vercel app_donate

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,16 +1,36 @@
-# Yieldgate
+# Yieldgate 🤝 Stake for Earth
 
 [![MIT license](https://img.shields.io/badge/License-MIT-blue.svg)](https://yieldgate.mit-license.org/)
-[![codecov](https://codecov.io/gh/yieldgate/yieldgate/branch/develop/graph/badge.svg?token=Y1BVBCAJK5)](https://codecov.io/gh/yieldgate/yieldgate)(Contracts)
+[![codecov](https://codecov.io/gh/yieldgate/yieldgate/branch/develop/graph/badge.svg?token=Y1BVBCAJK5)](https://codecov.io/gh/yieldgate/yieldgate)
```
Comment (sebastianst) [inline README.md:4]: I put that `(Contracts)` there to emphasize that this is the code coverage of the contract tests, not of the overall repo, incl. frontends. You think that's obvious?

File: packages/frontend-sfe/src/lib/truncateHash.ts [hunk 2]
```diff
@@ -1,3 +1,6 @@
-export const truncateHash = (hash: string, length = 38): string => {
-  return hash.replace(hash.substring(6, length), '...')
+export const truncateHash = (hash?: string, length = 38): string => {
+  if (!hash) return ''
+  let truncatedHash = hash
+  truncatedHash = truncatedHash.replace(truncatedHash.substring(6, length), '…')
+  return truncatedHash
```
Comment (sebastianst) [inline packages/frontend-sfe/src/lib/truncateHash.ts]: Is that necessary for making TS happy? Why can't you keep 
```suggestion
  return hash.replace(truncatedHash.substring(6, length), '…')
```
after the early exit? Wouldn't TS infer that `hash` is not undefined after the first line check?

File: packages/frontend-sfe/src/pages/api/newsletter/[...newsletter].ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
+import { env } from '@lib/environment'
```
Comment (sebastianst) [inline packages/frontend-sfe/src/pages/api/newsletter/[...newsletter].ts:1]: This file has a funky file name 😂

**Review by sebastianst — COMMENTED** [review]
Do you use file `packages/frontend-sfe/public/images/hero-illustration-draft-3.png`?

File: README.md (hunk 1 above)
Comment (wottpal) [reply README.md:4]: i just had the feeling it looked very bad, as the text is not aligned with the images very well. and everyone interested in coverage can just click on the badge 🤷‍♂️

File: packages/frontend-sfe/src/lib/truncateHash.ts (hunk 2 above)
Comment (wottpal) [reply packages/frontend-sfe/src/lib/truncateHash.ts]: Changed in https://github.com/yieldgate/yieldgate/pull/67/commits/397110aa208e37a69503ffb1997997c94eebbced.

In another iteration, iIdid two things with `truncateHash` on multiple lines. There it made sense to have a dedicated variable.

File: packages/frontend-sfe/src/pages/api/newsletter/[...newsletter].ts (hunk 3 above)
Comment (wottpal) [reply packages/frontend-sfe/src/pages/api/newsletter/[...newsletter].ts:1]: Haha true, I'm already quite used to this Next.js syntax. It just means to really catch all api-routes nested under `/api/newsletter/**`: https://nextjs.org/docs/api-routes/dynamic-api-routes#catch-all-api-routes (but not necessary here, as I'm not going deeper than level 1. but still, does not hurt.)

**Comment by wottpal** [comment]
> Do you use file `packages/frontend-sfe/public/images/hero-illustration-draft-3.png`?

nope, removed it ✅

File: README.md (hunk 1 above)
Comment (sebastianst) [reply README.md:4]: Also fine for me, it doesn't look too great misaligned

**Review by sebastianst — APPROVED** [review]

**Pull request merged**